calculator/src/main/java/com/market/banica/calculator/service/JMXServiceImpl.java

```diff
@@ -0,0 +1,299 @@
+        }
+    }
+
+    @Override
+    @ManagedOperation
+    public void addIngredient(String recipeName, String ingredientName, String quantityAsString) {
+        LOGGER.debug("JMX server impl: In addIngredient method");
+        LOGGER.info("AddIngredient called from JMX server with parameters recipeName {},ingredientName {} and quantityAsString {}",
+                recipeName, ingredientName, quantityAsString);
+
+        Product ingredient = retrieveProductFromDatabase(ingredientName);
+        Product recipe = retrieveProductFromDatabase(recipeName);
+
+        validateProductExist(ingredientName, ingredient);
+        validateProductExist(recipeName, recipe);
+
+        int quantity = getValueAsInt(quantityAsString);
+
+        recipe.getIngredients().put(ingredientName,quantity);
+
+        productService.createBackUp();
+
+        LOGGER.info("Ingredient added from JMX server for recipe {} and ingredient {} with value {}"
+                , recipeName, ingredientName, quantityAsString);
+    }
+
+    @Override
+    @ManagedOperation
+    public void setProductQuantity(String recipeName, String ingredientName, String newValue) {
+        LOGGER.debug("JMX server impl: In setProductQuantity method");
+        LOGGER.info("SetProductQuantity called from JMX server with parameters recipeName {},ingredientName {}" +
+                " and newValue {}", recipeName, ingredientName, newValue);
+
+        Product parentProduct = retrieveProductFromDatabase(recipeName);
+
+        validateProductExist(recipeName, parentProduct);
+
+        int newQuantity = getValueAsInt(newValue);
+
+        if (doesIngredientBelongToRecipe(ingredientName, parentProduct)) {
+
+            setProductQuantity(recipeName, parentProduct, newQuantity);
+
+            productService.createBackUp();
+        } else {
+
+            throwExceptionWhenProductDoesNotBelongToRecipe(recipeName, ingredientName);
+        }
+
+        LOGGER.info("Value set from JMX server for recipe {} and ingredient {} with value {}"
+                , recipeName, ingredientName, newValue);
+    }
+
+    @Override
+    @ManagedOperation
+    public String getProductQuantity(String recipeName, String ingredientName) {
+        LOGGER.debug("JMX server impl: In getProductQuantity method");
+        LOGGER.info("GetProductQuantity called from JMX server for recipe {} and ingredient {}", recipeName, ingredientName);
+
+        Product parentProduct = retrieveProductFromDatabase(recipeName);
+
+        validateProductExist(recipeName, parentProduct);
+
+        if (doesIngredientBelongToRecipe(ingredientName, parentProduct)) {
+
+            LOGGER.info("Value checked from JMX server for recipe {} and ingredient {}", recipeName, ingredientName);
+            return String.valueOf(getProductQuantity(ingredientName, parentProduct));
+
+        } else {
+
+            return throwExceptionWhenProductDoesNotBelongToRecipe(recipeName, ingredientName);
```

Comment:
No point of returning method that throws exception - switch the places of return quantity and throwExceptionWhenProductDoesNotBelongToRecipe

Reply:
Done